libs/core/include/core/reactor.hpp

```diff
@@ -59,16 +64,34 @@ class Reactor
   void StartWorker();
   void StopWorker();
   void Monitor();
+  void ReactorWatch();
 
   telemetry::HistogramPtr       CreateHistogram(char const *name, char const *description) const;
   telemetry::CounterPtr         CreateCounter(char const *name, char const *description) const;
   telemetry::GaugePtr<uint64_t> CreateGauge(char const *name, char const *description) const;
 
   std::string const name_;
   Flag              running_{false};
+  Flag              not_destructing_{true};
 
   RunnableMap work_map_{};
   ThreadPtr   worker_{};
+  ThreadPtr   watcher_{};
+
+  // Keeping track of the last item executed
+  uint32_t     execution_counter_{0};
```

Comment:
Are these plain uints always accessed under a locked mutex?

Reply:
execution_counter was unsafe - executions_*_too_long_ are only used for testing but I made them atomic to be thorough